### PR TITLE
Change threads to tree-like view in web UI

### PR DESCRIPTION
**Pull request by mumot1999**
Hello, I have created feature for simplifying view of responses for statuses.

Before: 
Screenshot from 2022-01-12 11-27-09

After:
Screenshot from 2022-01-12 11-27-49

### CODE REVIEW

**Comment by mumot1999** [comment]
Any tool for auto resolve codeclimate?

**Comment by ClearlyClaire** [comment]
Hello, thank you for your pull request!

I have not reviewed the code at all yet, so I'm only commenting based on the screenshots: this looks surprisingly good, and makes understanding forking threads easier. However, I am not completely sure how well it would work in practice with long threads. Furthermore, a very common usage pattern (maybe a bit less on Mastodon than Twitter thanks to the increased characters limit) is to post self-replies in a linear thread, which in this case would increase the depth on each post.

For these reasons I'm not sure this should be the only—or even default—way of rendering threads. But it is definitely worth investigating!

> Any tool for auto resolve codeclimate?

As far as I know, the CodeClimate warnings come from `eslint`. You could try `eslint --fix`, although I have never tried it myself.

Note also that not 100% of our existing code adheres to the coding styles checked by CodeClimate, so using `eslint --fix` is likely to produce changes in other parts of the code, and that should not be submitted as part of this PR.

File: app/javascript/mastodon/features/status/index.js [hunk 1]
```diff
@@ -514,8 +549,22 @@ class Status extends ImmutablePureComponent {
       ancestors = <div>{this.renderChildren(ancestorsIds)}</div>;
     }
 
+    const wrapDescendants = (elements) => {
```
[the remark this answers is not in the document]
Comment (mumot1999) [reply app/javascript/mastodon/features/status/index.js:552]: sorry for force push, missed I have added wrong file

**Comment by mumot1999** [comment]
Sorry for force push, missed I have added wrong file.

> I am not completely sure how well it would work in practice with long threads. 

I have added parameter for setting maximum tree depth. Currently it's hardcoded for 3.

What do you think about place a button in right corner for changing the view of threads. 
User could choose tree view or list.

**Comment by ClearlyClaire** [comment]
I'm trying it out on a real instance to get a sense of whether the design works well or not, and apparently, this does not work well on multi-column or mobile (single-column on a wide screen does not have this issue):
image

> What do you think about place a button in right corner for changing the view of threads.
> User could choose tree view or list.

That sounds like a good idea!

**Pull request closed without merging**